src/index.js

```diff
@@ -135,6 +134,65 @@ class IpfsRepo {
     })
   }
 
+  /**
+   * Returns the repo locker to be used. Null will be returned if no locker is requested
+   *
+   * @private
+   * @returns {Locker}
+   */
+  _getLocker () {
+    if (this.options.locker === false) {
```

Comment:
This way of options is :-1: from my side. This forces `locker` to be of multiple types, so disabling the locker should check for `!= null`  and be if it is not set, rather than a boolean

Reply:
I know multiple types for an option isn’t especially desirable, but it feels to me like just using `lock` for both custom and built-in types instead of adding a new option would be *much* nicer from the perspective of a user. Maybe we could move towards the type being a Locker object instead of a string? You could support built-ins with:

```js
// Expose the built-in collection as a frozen static var on Repo
new Repo({ lock: Repo.lockers.memory })
// OR
new Repo({ lock: require('ipfs-repo/lock-memory') })
```

…though we’d probably want to keep string support as a deprecated feature for a little while.

(Found my way here after reviewing ipfs/js-ipfs#1303; I hope I’m not butting in.)